src/sentry_options.c

```diff
@@ -396,6 +401,20 @@ sentry_options_set_logger(
     opts->logger.logger_data = userdata;
 }
 
+void
+sentry_options_set_logger_level(sentry_options_t *opts, sentry_level_t level)
+{
+    opts->logger.logger_level = level;
+    opts->debug = true;
```

Comment:
We implicitly expect that the user wants to have debug output when setting the logger level (hence we call this from `sentry_options_set_debug` as well).